controller/engine_controller.go

```diff
@@ -973,6 +973,24 @@ func (m *EngineMonitor) refresh(engine *longhorn.Engine) error {
 						volumeInfo.UnmapMarkSnapChainRemoved, engine.Spec.UnmapMarkSnapChainRemovedEnabled)
 				}
 			}
+
+			engine.Status.SnapshotMaxCount = volumeInfo.MaximumSnapshotCount
```

Comment:
Why we need to do this update? Shouldn't `engine.Status.SnapshotMaxCount` be synced eventually from the engine? The same question for `engine.Status.SnapshotMaxSize`.

Reply:
The `engine.Status.SnapshotMaxCount` means the current value in longhorn-engine. If it's not equal to `engine.Spec.SnapshotMaxCount`, we send a request to update it.